clients/src/test/java/org/apache/kafka/clients/consumer/internals/HeartbeatRequestManagerTest.java

```diff
@@ -251,23 +304,25 @@ public void testSkippingHeartbeat(final boolean shouldSkipHeartbeat) {
 
     @Test
     public void testTimerNotDue() {
-        mockStableMember();
         time.sleep(100); // time elapsed < heartbeatInterval, no heartbeat should be sent
         NetworkClientDelegate.PollResult result = heartbeatRequestManager.poll(time.milliseconds());
+
         assertEquals(0, result.unsentRequests.size());
         assertEquals(DEFAULT_HEARTBEAT_INTERVAL_MS - 100, result.timeUntilNextPollMs);
+
+        when(pollTimer.remainingMs()).thenReturn(1800L);
         assertEquals(DEFAULT_HEARTBEAT_INTERVAL_MS - 100, heartbeatRequestManager.maximumTimeToWait(time.milliseconds()));
 
         // Member in state where it should not send Heartbeat anymore
         when(subscriptions.hasAutoAssignedPartitions()).thenReturn(true);
```

Comment:
It is actually needed for this test. ```maximumTimeToWait()``` will always return 0 if ```pollTimer``` is a mock. Just did some testing turning ```pollTimer``` from a mock to a spy and it worked out pretty good. All tests are working, I think it may be beneficial to keep ```pollTimer``` a spy, since I notice in many places we ```verify``` on ```pollTimer```. Let me know what you think, my changes are pushed

Reply:
I can do this if I turn ```pollTimer``` into a spy, I tried that out and I think we should keep it. It makes the tests more strict on timing, since a mock would have to be stubbed to get the desired behavior. Let me know what you think